api/src/Controller/Info.php

```diff
@@ -46,6 +46,7 @@ public function index(): Response
           'default_lang' => $this->getParameter('lang'),
           'allow_email' => 'true' == $this->getParameter('allow.email'),
           'allow_public_links' => 'true' == $this->getParameter('allow.public.links'),
+          'allow_message_reactions' => 'true' == $this->getParameter('allow.message.reactions'),
```

Comment:
I like the possibility of disabling reactions but you've done it only on the backend side. `emoji-picker-react` is still going to be compiled as part of the webapp and pushed to the user.  
Ideally, the frontend would check if this parameter is enabled and only load the emoji picker code if it is but doing so is more complicated.

Reply:
The `import EmojiPicker from "emoji-picker-react"' is within a separate component. My understanding was that because this component is not loaded because of the `&& api?.info?.allow_message_reactions` check, this would mean `emoji-picker-react` wouldn't be pushed to the user?

I'm new to this so happy to take advice on making this better.

Reply:
It will not display it but the code is still here (and has to be downloaded and parsed by the client). `import` is resolved during transpilation (to create a javascript payload that will be send to the browser) and the `&& api?.info?.allow_message_reactions` is at runtime.  

That said, it's possible that parcel is doing some magic that I'm not aware of. I'll check exactly what's happening when trying your PR locally

Reply:
You were right, it's still bundled as part of the overall js package (to the tune of 250kb or so). I think I've got this sorted now though.

I've pushed a new commit that uses lazy loading. I came across a [bug in preact](https://github.com/preactjs/preact/issues/2519) so I have run `npm update` to update packages to latest minor versions, which has resolved the bug. The commit includes the updated package.json/lock files.

Through some trial and error I've worked out that importing on mount of the component (instead of waiting for first use) means the user doesn't have to mess around clicking the button a few times when they first try to select a reaction.

Again - I'm no expert, so please let me know if something needs changing.

Reply:
From my testing, it works ! Thanks a lot for that